src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessarySuppressionsDiagnosticAnalyzer.cs

```diff
@@ -66,21 +76,43 @@ public void AnalyzeAssemblyOrModuleAttribute(SyntaxNode attributeSyntax, Semanti
                     return;
                 }
 
-                DiagnosticDescriptor rule;
                 if (SuppressMessageAttributeState.HasInvalidScope(namedAttributeArguments, out var targetScope))
                 {
-                    rule = s_invalidScopeDescriptor;
+                    reportDiagnostic(Diagnostic.Create(s_invalidScopeDescriptor, attributeSyntax.GetLocation()));
+                    return;
                 }
-                else if (_state.HasInvalidOrMissingTarget(namedAttributeArguments, targetScope))
+
+                if (_state.HasInvalidOrMissingTarget(namedAttributeArguments, targetScope, out var targetHasDocCommentIdFormat,
+                        out var targetSymbolString, out var targetValueOperation, out var resolvedSymbols))
                 {
-                    rule = s_invalidOrMissingTargetDescriptor;
+                    reportDiagnostic(Diagnostic.Create(s_invalidOrMissingTargetDescriptor, attributeSyntax.GetLocation()));
+                    return;
                 }
-                else
+
+                if (resolvedSymbols.Length > 0 && !targetHasDocCommentIdFormat)
                 {
+                    RoslynDebug.Assert(!string.IsNullOrEmpty(targetSymbolString));
+                    RoslynDebug.Assert(targetValueOperation != null);
+
+                    var properties = ImmutableDictionary<string, string?>.Empty;
+                    if (resolvedSymbols.Length == 1)
+                    {
+                        // We provide a code fix for the case where the target resolved to a single symbol.
+                        var docCommentId = DocumentationCommentId.CreateDeclarationId(resolvedSymbols[0]);
+                        if (!string.IsNullOrEmpty(docCommentId))
+                        {
+                            // Suppression target has an optional "~" prefix to distinguish it from legacy FxCop suppressions.
+                            // IDE suppression code fixes emit this prefix, so we we also add this prefix to new suppression target string.
+                            properties = properties.Add(DocCommentIdKey, "~" + docCommentId);
+                        }
+                    }
+
+#pragma warning disable CS8620 // Mismatch in nullability of 'properties' parameter and argument types - Parameter type for 'properties' has been updated to 'ImmutableDictionary<string, string?>?' in newer version of Microsoft.CodeAnalysis (3.7.x).
```

Comment:
can we just `!` this away?  that way we can also get a notification about an unnecessary `!` that should be removed later on.